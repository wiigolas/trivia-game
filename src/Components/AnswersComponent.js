import React, {Component} from 'react';
import { Alert, Button, View, Text, TouchableHighlight } from 'react-native';

import styles from '../Styles/styles';

export default class AnswersComponent extends Component<Props> {
  constructor(props) {
    super(props);
    this.state = {
      answers: {
        correctAnswers: 0,
        incorrectAnswers: 0,
      }
    };

    this.checkAnswer = this.checkAnswer.bind(this);
  }

  answers = () => {
    if(this.props.incorrectAnswers){
      const array =  this.props.incorrectAnswers.concat([this.props.correctAnswer]);
      const shuffledArray = this.shuffle(array);
      return shuffledArray;
    }else{
      return [];
    }
  }

  checkAnswer(selectedAnswer) {
    if(selectedAnswer == this.props.correctAnswer) {
      this.setState({
        answers: {
          correctAnswers: this.state.answers.correctAnswers + 1,
          incorrectAnswers: this.state.answers.incorrectAnswers,
        }
      });
    } else {
      this.setState({
        answers: {
          correctAnswers: this.state.answers.correctAnswers,
          incorrectAnswers: this.state.answers.incorrectAnswers + 1,
        }
      });
    };
    this.props.totalAnswers(this.state.answers);
    this.props.nextQuestion();

  };

  shuffle(array) {
    for (let i = array.length - 1; i > 0; i--) {
        const j = Math.floor(Math.random() * (i + 1));
        [array[i], array[j]] = [array[j], array[i]];
    }
    return array;
  }
  render(){
    return (
      <View style={styles.answersContainer}>
        {
          this.answers().map((answer, key) => {
            return (
              <TouchableHighlight
                key={key}
                style={styles.answerSelectButton}
                onPress={() => this.checkAnswer(answer) }
              >
                <Text style={styles.textStyle}>{answer}</Text>
              </TouchableHighlight>
            );
          })
        }
      </View>
    )
  }
}
