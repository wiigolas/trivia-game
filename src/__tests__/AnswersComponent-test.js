import 'react-native';
import React from 'react';
import { configure, shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createSerializer } from 'enzyme-to-json';
import sinon from 'sinon';

expect.addSnapshotSerializer(createSerializer({ mode: 'deep'}));

configure({ adapter: new Adapter() });

import AnswersComponent from '../Components/AnswersComponent';

describe('testing the <AnswersComponentComponent>', () => {
  const incorrectAnswers = [
    'answer',
    'answer',
    'answer',
  ];
  const correctAnswer = 'answer';
  it('renders correctly', () => {
    const wrapper = shallow(
      <AnswersComponent
      incorrectAnswers={incorrectAnswers}
      correctAnswer={correctAnswer}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
})
