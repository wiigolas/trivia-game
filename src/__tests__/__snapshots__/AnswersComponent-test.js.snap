// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the <AnswersComponentComponent> renders correctly 1`] = `
<Component>
  <Button
    key="0"
    onPress={[Function]}
    title="answer"
  />
  <Button
    key="1"
    onPress={[Function]}
    title="answer"
  />
  <Button
    key="2"
    onPress={[Function]}
    title="answer"
  />
  <Button
    key="3"
    onPress={[Function]}
    title="answer"
  />
</Component>
`;
